src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,165 @@
+#include "AxialGeoMultiscaleMapping.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include "mesh/Mesh.hpp"
+
+namespace precice::mapping {
+
+AxialGeoMultiscaleMapping::AxialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    MultiscaleAxis axis,
+    double         radius)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type),
+      _axis(axis),
+      _radius(radius)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void AxialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == SPREAD) {
+      PRECICE_CHECK(input()->vertices().size() == 1, "You can only define an axial geometric multiscale mapping of type spread from a mesh with exactly one vertex.");
+
+      // TODO: get rid of hardcoded values via access to mesh dimension
+      const int inValueDimensions  = 1;
+      const int outValueDimensions = 3;
```

Comment:
This will have to be changed in accordance with #1742

Reply:
I updated these lines to extract the dimensions of each mesh, but we don't yet support 1D meshes.

Reply:
We have a separate issue for that, and we decided to merge this PR before addressing the 1D mesh support issue: https://github.com/precice/precice/issues/1669